internal/controllers/lvmcluster/controller.go

```diff
@@ -253,128 +240,57 @@ func (r *Reconciler) reconcile(ctx context.Context, instance *lvmv1alpha1.LVMClu
 	if len(errs) > 0 {
 		err := fmt.Errorf("failed to reconcile resources managed by LVMCluster: %w", errors.Join(errs...))
 		r.WarningEvent(ctx, instance, EventReasonErrorResourceReconciliationFailed, err)
+		statusErr := r.updateLVMClusterStatus(ctx, instance, err)
+		if statusErr != nil {
+			logger.Error(statusErr, "failed to update LVMCluster status")
+		}
 		return ctrl.Result{}, err
 	}
 
 	msg := "successfully reconciled LVMCluster"
 	logger.Info(msg, "resourceSyncElapsedTime", resourceSyncElapsedTime)
 	r.NormalEvent(ctx, instance, EventReasonResourceReconciliationSuccess, msg)
+	statusErr := r.updateLVMClusterStatus(ctx, instance, nil)
+	if statusErr != nil {
+		return ctrl.Result{}, statusErr
+	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *Reconciler) updateLVMClusterStatus(ctx context.Context, instance *lvmv1alpha1.LVMCluster) error {
+func (r *Reconciler) updateLVMClusterStatus(ctx context.Context, instance *lvmv1alpha1.LVMCluster, reconcileError error) error {
 	logger := log.FromContext(ctx)
 
-	vgNodeMap := make(map[string][]lvmv1alpha1.NodeStatus)
+	if reconcileError != nil {
```

Comment:
this if statement is not necessary if you simply set the resourcesAvailableCondition at line 241, then you can also drop passing the error and rename the function to setVolumeGroupsReadyCondition